runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -151,6 +152,8 @@ public void write(IToken token, Object data) throws OverwriteException {
                 // to validate the state of the write, we know that it didn't went through.
                 if (re instanceof OverwriteException) {
                     throw re;
+                } else if (re instanceof WriteTooLarge) {
+                    throw re;
                 }
 
                 validateStateOfWrittenEntry(token.getTokenValue(), ld);
```

Comment:
@rogermichoud this method is problematic, it will throw an Overwrite exception regardless of the exception type. Please look into it.